rsciio/_hierarchical.py

```diff
@@ -52,9 +52,20 @@ def flatten_data(x):
 
 
 def unflatten_data(data, shape):
+    try:
+        # Since h5py doesn't support numpy unicode dtype, we need to save
+        # data with h5py variable length str dtype and when reading the
+        # ragged data, we need to convert it back to numpy unicode dtype
+        if data.dtype.metadata["vlen"].metadata["vlen"] == str:
```

Comment:
I am not sure how robust it would be but this is the only way (workaround) that I could find to figure out if the dtype of the ragged are string.

Reply:
Yeah, I am not keen on this but I would ok with keeping it knowing that I couldn't find any another solution with h5py, even if there is a risk that it breaks at some point because this attribute is "weakly" documented.
This is another justification for consider the alternative approach more seriously.